tools/db_bench_tool.cc

```diff
@@ -4560,9 +4564,29 @@ void VerifyDBFromDB(std::string& truth_db_name) {
     std::unique_ptr<const char[]> key_guard;
     Slice key = AllocateKey(&key_guard);
 
+    std::unique_ptr<const char[]> upper_bound_key_guard;
+    Slice upper_bound = AllocateKey(&upper_bound_key_guard);
+    std::unique_ptr<const char[]> lower_bound_key_guard;
+    Slice lower_bound = AllocateKey(&lower_bound_key_guard);
+
     Duration duration(FLAGS_duration, reads_);
     char value_buffer[256];
     while (!duration.Done(1)) {
+      uint64_t seek_pos = thread->rand.Next() % FLAGS_num;
+      GenerateKeyFromInt(seek_pos, FLAGS_num, &key);
+      if (FLAGS_max_scan_distance != 0) {
+        if (FLAGS_reverse_iterator) {
+          GenerateKeyFromInt(std::max(0ul, seek_pos - FLAGS_max_scan_distance),
```

Comment:
I think the signed one (`FLAGS_max_scan_distance`) gets converted to unsigned (`seek_pos`) then the subtraction can underflow. If this is right I wonder why the compiler doesn't complain.

Reply:
I'm not sure why it's not complaining about underflow, but it's definitely unsigned (otherwise the compiler would have complained about comparing it to `0ul`). I'll make the max comparison use signed ints and cast the result to unsigned. Thanks for catching it!